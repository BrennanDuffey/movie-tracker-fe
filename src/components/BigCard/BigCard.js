import React, {Component} from 'react';
import {connect} from 'react-redux';
import {grabCurrMovie} from '../../actions';

class BigCard extends Component{
  constructor(props){
    super(props)
    this.state={

    }
  }

  componentDidMount() {
    console.log(this.props.movie)
  }

  render() {
    let {title, poster, backdrop, summary, genres, rating, release, isFavorite} = this.props.movie
    return(
      <section className="BigCard">
        <article className="big-main">
          { title && 
            <h2>{title} 
              <span>({release})</span>
            </h2>}
          { summary && <p>{summary}</p> }
          { genres && <p>{genres}</p>}
        </article>
        { rating && <h2>{rating}</h2>}
        { poster && <img src={poster} alt={title} />}
        { backdrop && <img src={backdrop} alt="Backdrop Unavailable" />}
        { isFavorite && <i>Heart</i>}
      </section>
    )
  }
}

const mapStateToProps =(state)=>({
  movie: state.currMovie
});

const mapDispatchToProps =(dispatch)=>({
  grabCurrMovie: ({})=> dispatch(grabCurrMovie({}))
});

export default connect(mapStateToProps, mapDispatchToProps)(BigCard); 