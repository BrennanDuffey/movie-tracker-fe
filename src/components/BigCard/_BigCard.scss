.BigCard {
  background-color: teal;
  position: relative;
  z-index: 15;

  h2 {
    font-size: 40px;
    color: magenta;
  }
}
