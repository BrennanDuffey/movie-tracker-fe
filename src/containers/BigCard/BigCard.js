import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {NavLink, Redirect} from 'react-router-dom';
import { toggleFavorite, errorMessage } from "../../actions";
import {
  deleteFavoriteFetch,
  agnosticFetch
} from "../../utils/apiCalls/apiCalls";

export class BigCard extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  handleFavorite = e => {
    e.preventDefault();
    console.log(this.props.movie);
    let { isFavorite } = this.props.movie;
    if (!isFavorite) {
      console.log('add')
      this.addFavorite();
    } else {
      console.log("remove");
      this.removeFavorite();
    }
  };

  removeFavorite = async () => {
    let { user, toggleFavorite, setErrorMessage } = this.props;
    let { id } = this.props.movie;
    try {
      const response = deleteFavoriteFetch(user.id, id);
      toggleFavorite(id);
    } catch {
      setErrorMessage("Error in deleting favorite");
    }
  };

  addFavorite = async () => {
    let { user, toggleFavorite, setErrorMessage } = this.props;
    let {
      title,
      id,
      summary,
      poster,
      release,
      rating
    } = this.props.movie;
    const url = "http://localhost:3000/api/users/favorites/new";
    const init = {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        title: title,
        user_id: user.id,
        movie_id: id,
        overview: summary,
        poster_path: poster,
        release_date: release,
        vote_average: rating
      })
    };
    try {
      const result = await agnosticFetch(url, init);
      toggleFavorite(id);
    } catch {
      setErrorMessage("Please sign in to favorite");
      window.alert(this.props.errorMessage);
    }
  };

  render() {
    let {
      title,
      backdrop,
      summary,
      genres,
      rating,
      release,
      isFavorite,
    } = this.props.movie;
    let favClass = `${isFavorite}`;

    if (this.props.movie === {}) {
      return <Redirect to="/" />;
    }

    return (
      <main>
        <section className="BigCard">
          {backdrop && (
            <img
              className="backdrop"
              src={backdrop}
              alt="Backdrop Unavailable"
            />
          )}
          <article className="big-content">
            {title && <h2>{title}</h2>}
            {release && <h4>{release}</h4>}
            <div className="big-subcontent">
              {summary && <p>{summary}</p>}
              {genres && <p>{genres}</p>}
            </div>
            {rating && <h3>Avg Score: {rating}</h3>}
          </article>
          <button className="faveBtn" onClick={this.handleFavorite}>
            <i className={`fas fa-star fave-${favClass}`} />
          </button>
          <NavLink to="/" className="close">
            <i className="fas fa-times-circle" />
          </NavLink>
        </section>
      </main>
    );
  }
}

BigCard.propTypes = {
  movie: PropTypes.object,
  user: PropTypes.object
};

const mapStateToProps = state => ({
  movie: state.currMovie,
  user: state.user
});

const mapDispatchToProps = dispatch => ({
  setErrorMessage: message => dispatch(errorMessage(message)),
  toggleFavorite: id => dispatch(toggleFavorite(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(BigCard); 