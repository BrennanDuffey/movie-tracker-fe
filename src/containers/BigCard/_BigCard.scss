main {
  position: relative;
  box-sizing: border-box;
  width: 90vw;
  height: min(100vh);
  display: grid;
  grid-template-rows: 10% 90%;
  background-color: #3a3a3a;
  background: url('https://images.unsplash.com/photo-1489599849927-2ee91cede3ba?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80');
  background-size: cover;
  backdrop-filter: 80%;
}

.BigCard {
  background-color: #500000;
  border-radius: 30px;
  position: absolute;
  top: 25%;
  left: 15%;
  width: 60vw;
  height: 50vh;
  z-index: 15;
  color: silver;
  box-shadow: 5px 5px 10px black;


  h2 {
    font-size: 28px;
    margin: 0;
  }

  .backdrop {
    position: absolute;
    width: 50%;
    height: 100%;
    z-index: 2;
    border-right: 12px outset #690e0eb7;
    border-radius: 0px 120px 120px 0px;
  }

  .big-content{
    position: absolute;
    z-index: 10;
    width: 40%;
    right: 5%;
    top: 15%;
  }

  .big-subcontent {
    border-top: 3px solid silver;
    border-bottom: 3px solid silver;
    max-height: 150px;
    overflow: scroll;
  }

  .close {
    display: flex;
    justify-content: center;
    align-items: center;
    position: absolute;
    right: 5%;
    top: 5%;
    text-decoration: none;
    color: black;
    font-size: 27px;
    text-shadow: .5px .5px 5px gray;
    height: 30px; width: 30px;
    border-radius: 50%;
  }

  .faveBtn {
    position: absolute;
    right: 5%;
    bottom:5%;
    font-size: 27px;
    z-index: 15;
    color: silver;
    background: none;
    border: none;

    &:hover {
      color: rgb(232, 181, 27);
    }
  }

  .fave-true {
  color: rgb(232, 181, 27);

  &:hover {
    color: rgb(91, 91, 91);
  }
}
}


