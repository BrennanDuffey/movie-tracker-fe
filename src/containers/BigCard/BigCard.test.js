import {BigCard} from './BigCard';
import React from 'react';
import { shallow } from 'enzyme';

let mockMovie = {
  title: "movie",
  poster: "http",
  backdrop: "http",
  summary: "words",
  genres: ["action"],
  isFavorite: false,
  release: 2000,
  rating: 9.9,
  id: 69
};

describe('BigCard', () => {
  let wrapper;

  beforeEach(()=>{
    wrapper = shallow(
      <BigCard movie={mockMovie} />
    );
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('shouldnt render if there is no movie object', ()=>{
    wrapper = shallow(
      <BigCard movie={{}}/>
    );
  });

  
});