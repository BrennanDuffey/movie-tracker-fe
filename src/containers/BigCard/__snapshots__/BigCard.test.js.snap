// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigCard should match snapshot 1`] = `
<main>
  <section
    className="BigCard"
  >
    <img
      alt="Backdrop Unavailable"
      className="backdrop"
      src="http"
    />
    <article
      className="big-content"
    >
      <h2>
        movie
      </h2>
      <h4>
        2000
      </h4>
      <div
        className="big-subcontent"
      >
        <p>
          words
        </p>
        <p>
          action
        </p>
      </div>
      <h3>
        Avg Score: 
        9.9
      </h3>
    </article>
    <button
      className="faveBtn"
      onClick={[Function]}
    >
      <i
        className="fas fa-star fave-false"
      />
    </button>
    <NavLink
      className="close"
      to="/"
    >
      <i
        className="fas fa-times-circle"
      />
    </NavLink>
  </section>
</main>
`;
